pallets/vault-registry/src/tests.rs

```diff
@@ -1711,3 +1712,146 @@ fn test_offchain_worker_unsigned_transaction_submission() {
 		);
 	})
 }
+
+#[test]
+fn integration_single_vault_receives_per_block_reward() {
+	run_test(|| {
+		//set up reward values
+		let reward_per_block: u64 = 1000u64;
+		assert_ok!(ext::staking::add_reward_currency::<Test>(DEFAULT_WRAPPED_CURRENCY));
+		assert_ok!(<reward_distribution::Pallet<Test>>::set_reward_per_block(
+			RawOrigin::Root.into(),
+			reward_per_block.into()
+		));
+		//register vault and issue tokens.
+		let issue_tokens: u128 = DEFAULT_COLLATERAL / 10 / 2; // = 5
+		let id = create_vault_and_issue_tokens(issue_tokens, DEFAULT_COLLATERAL, DEFAULT_ID);
+		assert_eq!(
+			<Test as fee::Config>::VaultRewards::get_stake(&id.collateral_currency(), &id),
+			Ok(DEFAULT_COLLATERAL)
+		);
+		assert_eq!(<pallet_balances::Pallet<Test>>::free_balance(&id.account_id), 0u128);
+
+		//distribute fee rewards
+		<reward_distribution::Pallet<Test>>::execute_on_init(2u32.into());
+		//collect rewards
+		let origin = RuntimeOrigin::signed(id.account_id);
+		assert_ok!(<reward_distribution::Pallet<Test>>::collect_reward(
+			origin.into(),
+			id.clone(),
+			DEFAULT_NATIVE_CURRENCY,
+			None,
+		));
+		assert_eq!(
+			<pallet_balances::Pallet<Test>>::free_balance(&id.account_id),
+			reward_per_block.into()
+		);
+	});
+}
+
+#[test]
+fn integration_multiple_vault_same_collateral_per_block_reward() {
+	run_test(|| {
+		//set up reward values
+		let reward_per_block: u64 = 100000u64;
+		assert_ok!(ext::staking::add_reward_currency::<Test>(DEFAULT_WRAPPED_CURRENCY));
+		assert_ok!(<reward_distribution::Pallet<Test>>::set_reward_per_block(
+			RawOrigin::Root.into(),
+			reward_per_block.into()
+		));
+
+		//register vaults and issue tokens.
+		let issue_tokens: u128 = 5;
+
+		let collateral_vault_1 = 1000u128;
+		let id = create_vault_and_issue_tokens(issue_tokens, collateral_vault_1, DEFAULT_ID);
+		assert_eq!(
+			<Test as fee::Config>::VaultRewards::get_stake(&id.collateral_currency(), &id),
+			Ok(collateral_vault_1)
+		);
+
+		let collateral_vault_2 = 5000u128;
+		let id2 = create_vault_and_issue_tokens(issue_tokens, collateral_vault_2, OTHER_ID);
+		assert_eq!(
+			<Test as fee::Config>::VaultRewards::get_stake(&id2.collateral_currency(), &id2),
+			Ok(collateral_vault_2)
+		);
+
+		//distribute fee rewards
+		<reward_distribution::Pallet<Test>>::execute_on_init(2u32.into());
+		//collect rewards
+		let origin = RuntimeOrigin::signed(id.account_id);
+		assert_ok!(<reward_distribution::Pallet<Test>>::collect_reward(
+			origin.into(),
+			id.clone(),
+			DEFAULT_NATIVE_CURRENCY,
+			None,
+		));
+
+		//expected value  = (1000*5/(5000*5 + 1000*5))*100000 = 16666
```

Comment:
That is to get explicit the price in USD, ([value in usd is defined here](https://github.com/pendulum-chain/spacewalk/pull/428/files#diff-d6cdd749af964b12aacc03febe5032c67c92157c7f9864bdaec757186d686165R273). It is confusing that the issue is also 5, I will change that so there is no confusion since as you say, it does not matter in this example.